src/main/java/org/jabref/gui/openoffice/OOBibBase.java

```diff
@@ -872,11 +889,86 @@ public void guiActionUpdateDocument(List<BibDatabase> databases, OOStyle style)
+
+                if (testDialog(errorTitle,
+                        fcursor.asVoidResult(),
+                        checkIfOpenOfficeIsRecordingChanges(doc))) {
+                    return;
+                }
+
+                try {
+                    UnoUndo.enterUndoContext(doc, "Create CSL bibliography");
+
+                    CSLCitationOOAdapter adapter = new CSLCitationOOAdapter(doc);
+                    adapter.readExistingMarks();
+
+                    // Collect only cited entries from all databases
+                    List<BibEntry> citedEntries = new ArrayList<>();
+                    for (BibDatabase database : databases) {
+                        for (BibEntry entry : database.getEntries()) {
+                            if (adapter.isCitedEntry(entry)) {
+                                citedEntries.add(entry);
+                            }
+                        }
+                    }
+
+                    // If no entries are cited, show a message and return
+                    if (citedEntries.isEmpty()) {
+                        dialogService.showInformationDialogAndWait(
+                                Localization.lang("Bibliography"),
+                                Localization.lang("No cited entries found in the document.")
+                        );
+                        return;
+                    }
+
+                    // Create a new cursor at the end of the document for bibliography insertion
+                    XTextCursor bibliographyCursor = doc.getText().createTextCursor();
+                    bibliographyCursor.gotoEnd(false);
+
+                    // Insert bibliography title
+                    String bibliographyTitle = Localization.lang("\nReferences");
+                    bibliographyCursor.setString("\n");
+
+                    // Apply formatting to the title (here, we make it bold and larger)
+                    XTextRange titleRange = bibliographyCursor.getStart();
+                    titleRange.setString(bibliographyTitle);
+                    XPropertySet titleProps = UnoRuntime.queryInterface(XPropertySet.class, titleRange);
+                    titleProps.setPropertyValue("CharWeight", com.sun.star.awt.FontWeight.BOLD);
+                    titleProps.setPropertyValue("CharHeight", 16f);
+
+                    // Move cursor to prevent reference mark bleed-out (current implementation) TODO: Remove once bleed-out is fixed
+                    bibliographyCursor.goRight((short) 1, false);
+
+                    // Insert bibliography entries
+                    // BibDatabaseContext bibDatabaseContext = new BibDatabaseContext(databases.getFirst());
+                    // TODO: Ask Chris if injecting database context is a good idea
+                    BibDatabaseContext bibDatabaseContext = Injector.instantiateModelOrService(BibDatabaseContext.class);
```

Comment:
It's a bad Idea.

The injector currently cares about singletons across the whole application. So injecting a BibDatabaseContext would assume, that there is only one BibDatabaseContext in the whole application. But there are multiple. one for each opened BibDatabase.